spec/classes/mod/cache_disk_spec.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'apache::mod::cache_disk', type: :class do
+  context 'on a Debian OS' do
+    include_examples 'Debian 11'
+
+    let(:params) do
+      {
+        cache_enable: ['/'],
+      }
+    end
+
+    let :pre_condition do
+      'class{ "apache":
+        default_mods   => ["cache"],
+        mod_dir        => "/tmp/junk",
+       }'
+    end
+
+    it { is_expected.to compile }
+    it { is_expected.to contain_class('apache::mod::cache_disk') }
+    it { is_expected.to contain_class('apache::mod::cache').that_comes_before('Class[Apache::Mod::Cache_disk]') }
+    it { is_expected.to contain_apache__mod('cache_disk') }
+
+    default_config = %r{CacheEnable disk /\nCacheRoot "/var/cache/apache2/mod_cache_disk"}
+
+    it {
+      expect(subject).to contain_file('cache_disk.conf')
+                           .with(content: default_config)
+    }
```

Comment:
Not sure about the suggestion from rubocop :upside_down_face: 

The line length feel reasonable with current editors and screen so would not mind seeing everything on the same line:

```suggestion
    it { is_expected.to contain_file('cache_disk.conf').with(content: default_config) }
```


Note that I replaced `expect(subject)` by `is_expected`: it does the same but makes the whole test [almost] readable as a sentence.